.github/workflows/ruby-build.yml

```diff
@@ -227,3 +235,56 @@ jobs:
         shell: bash
         run: |
           codeql database analyze --search-path "${{ runner.temp }}/ruby-bundle" --format=sarifv2.1.0 --output=out.sarif ../database ruby-code-scanning.qls
+
+  # This is a copy of the 'test' job that runs in a centos7 container.
+  # This tests that the extractor works correctly on systems with an old glibc.
+  test-centos7:
+    defaults:
+      run:
+        working-directory: ${{ github.workspace }}
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-latest
+    container:
+      image: centos:centos7
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs: [package]
+    steps:
+      - name: Install gh cli
+        run: |
+          yum-config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo
+          # fetch-codeql requires unzip and jq
+          # jq is available in epel-release (https://docs.fedoraproject.org/en-US/epel/)
+          yum install -y gh unzip epel-release
+          yum install -y jq
+      - uses: actions/checkout@v3
+      - name: Fetch CodeQL
+        uses: ./.github/actions/fetch-codeql
+
+      # Due to a bug in Actions, we can't use runner.temp here.
```

Comment:
Are you sure that `${{ runner.temp }}` does not work? It looks like things are supposed to be fixed in https://github.com/actions/runner/pull/1762 .

Reply:
Looking at the failed run https://github.com/github/codeql/actions/runs/4442367661/jobs/7798560621 it seems the bug was only half-fixed ... 

The use of `${{ github.temp }}` in the `with` clause of the download step was expanded correctly: `https://github.com/github/codeql/actions/runs/4442367661/jobs/7798560621
```
  with:
    name: codeql-ruby-bundle
    path: /home/runner/work/_temp
```
but the variable was expanded wrongly in the `run:` step:
```
unzip -q -d "/home/runner/work/_temp/ruby-bundle" "/home/runner/work/_temp/codeql-ruby-bundle.zip"
```